src/HealthGPS.Console/model_parser.cpp

```diff
@@ -59,10 +59,31 @@ hgps::RiskFactorSexAgeTable load_risk_factor_expected(const host::Configuration
     return table;
 }
 
+std::unique_ptr<hgps::DummyModelDefinition>
+load_dummy_model_definition(hgps::RiskFactorModelType type, const poco::json &opt) {
+    MEASURE_FUNCTION();
+
+    // Get dummy model parameters
+    std::vector<hgps::core::Identifier> names;
+    std::vector<double> values;
+    std::vector<double> policy;
+    for (const auto &[key, json_params] : opt["ModelParameters"].items()) {
+        names.emplace_back(key);
+        values.emplace_back(json_params["Value"].get<double>());
+        policy.emplace_back(json_params["Policy"].get<double>());
+    }
+
+    return std::make_unique<hgps::DummyModelDefinition>(type, std::move(names), std::move(values),
+                                                        std::move(policy));
+}
+
 std::unique_ptr<hgps::RiskFactorModelDefinition>
 load_static_risk_model_definition(const std::string &model_name, const poco::json &opt,
                                   const host::Configuration &config) {
     // Load this static model with the appropriate loader.
+    if (hgps::core::case_insensitive::equals(model_name, "dummy")) {
```

Comment:
Would it be useful for readability purposes - and maybe even for performance, I don't know - to turn this into a switch statement?

```cpp
switch(model_name) {
    case "dummy":
        return load_dummy_model_definition(hgps::RiskFactorModelType::Static, opt);
    case "hlm":
        return load_hlm_risk_model_definition(opt);
    case "staticlinear":
        return load_staticlinear_risk_model_definition(opt, config);
    default:
         throw hgps::core::HgpsException{
             fmt::format("Static model name '{}' not recognised", model_name)};
         }
}
```

Also, I would require that the model name is properly provided as lowercase. Allowing to give "Staticlinear", "StaTicLinEar", "staticLinear", etc. is just a mess and affect readability. If we want to enable that, let's make a conversion lowercase once when reading the configuration from JSON and move on assuming only lowercase thereafter.

Reply:
> Would it be useful...

I agree it would tidy it up a bit. Unfortunately C++ has other ideas: https://stackoverflow.com/questions/650162/why-cant-the-switch-statement-be-applied-to-strings

Ultimately agree about case. I've bought it up before somewhere also. Not sure how big a job it is to go through all the code and remove case insensitivity. I'll make an issue.